source/android/adaptivecards/src/main/java/com/microsoft/adaptivecards/renderer/registration/CardRendererRegistration.java

```diff
@@ -68,7 +68,7 @@ public static CardRendererRegistration getInstance()
 
     public void registerRenderer(CardElementType cardElementType, IBaseCardElementRenderer renderer)
     {
-        if (cardElementType != null || cardElementType == CardElementType.Unsupported)
+        if (cardElementType != null && cardElementType == CardElementType.Unsupported)
```

Comment:
>&& [](start = 36, length = 2)

It should be || and not &&.  However, the first check should be == and not !=

Reply:
I have a follow up PR that will include this fix plus passing string as id instead of CardElementType enum since this is final class and cannot be inherited.

---
In reply to: [140910916](https://github.com/Microsoft/AdaptiveCards/pull/684#discussion_r140910916) [](ancestors = 140910916)